### PR TITLE
General Doxygen Fixes

### PR DESCRIPTION
**Pull request by fsimonis**
I started to do some fixes in the Doxygen.
Also @uekerman and I started to revise the SolverInterface.

### CODE REVIEW

File: src/precice/bindings/fortran/SolverInterfaceFortran.hpp [hunk 1]
```diff
@@ -1,7 +1,7 @@
 #pragma once
 
-/**
- * @file This file contains a Fortran 77 compatible interface written in C/C++.
+/** @file
+ * This file contains a Fortran 77 compatible interface written in C/C++.
  *
  * It has been tested with: gfortran 4.4.3, ifort 12.1.0
```
Comment (uekerman) [inline src/precice/bindings/fortran/SolverInterfaceFortran.hpp]: This is most probably outdated. I would suggest to delete this information here. Such information should go to the wiki.

File: src/precice/SolverInterface.hpp [hunk 2]
```diff
@@ -131,45 +132,46 @@ class SolverInterface
...
    * @brief Initializes coupling data.
    *
-   * When in a coupled simulation an implicit coupling scheme is used, the
-   * starting values for the coupling data are assumed to be zero by default. If
-   * this is not the desired behavior, this method can be used to specify values
-   * different from zero using the write data methods. Only the first participant
-   * of the coupled simulation has to call this method, the second participant
-   * receives the values on calling initialize(). For parallel coupling, values in
-   * both directions are exchanged. Both participants need to call initializeData then.
+   * In an implicit coupling scheme, the starting values for the coupling data are zero by default.
```
Comment (uekerman) [inline src/precice/SolverInterface.hpp]:
```suggestion
   * The starting values for coupling data are zero by default.
```

File: src/precice/SolverInterface.hpp [hunk 3]
```diff
@@ -131,45 +132,46 @@ class SolverInterface
...
+   *
+   * \par Serial Coupling Scheme
+   * Only the first participant has to call this method, the second participant
+   * receives the values on calling initialize().
+   *
+   * \par Parallel Coupling Scheme
+   * Values in both directions are exchanged.
+   * Both participants need to call initializeData().
    *
    * @pre initialize() has been called successfully.
```
Comment (uekerman) [inline src/precice/SolverInterface.hpp]:
```suggestion
   * @pre initialize() has been called successfully.
   * @pre The action WriteInitialData is required 
```

Comment (uekerman) [reply src/precice/SolverInterface.hpp]: Forget this change. This is wrong. My mistake.

File: src/precice/SolverInterface.hpp [hunk 4]
```diff
@@ -264,20 +315,31 @@ class SolverInterface
...
   bool isActionRequired ( const std::string& action );
 
   /**
-   * @brief Tells preCICE that a required action has been fulfilled by a solver.
+   * @brief Indicates preCICE that a required action has been fulfilled by a solver.
    *
-   * For more details see method requireAction().
+   * @pre The solver fulfilled the specified action.
+   *
+   * @param[in] action the name of the action
```
Comment (floli) [inline src/precice/SolverInterface.hpp]: Typo: `@param[in}`

File: src/precice/MeshHandle.hpp [hunk 5]
```diff
@@ -36,6 +36,10 @@ class VertexIterator
 
   ~VertexIterator();
 
+  /** Constructs a vertex iterator from a given group.
+   * @param[IN] content the group of a mesh
```
[the remark this answers is not in the document]
Comment (fsimonis) [reply src/precice/MeshHandle.hpp]: I fixed all occurrences in the file.

File: src/precice/SolverInterface.hpp (hunk 4 above)
Comment (fsimonis) [reply src/precice/SolverInterface.hpp]: Fixed. thanks!

**Comment by fsimonis** [comment]
This PR is ready for review.

I will finish the `mapReadDataTo` and `mapWriteDataFrom` in another PR.

File: src/precice/bindings/fortran/SolverInterfaceFortran.hpp (hunk 1 above)
Comment (fsimonis) [reply src/precice/bindings/fortran/SolverInterfaceFortran.hpp]:
```suggestion
```

**Pull request merged**